### PR TITLE
Concurrency / it-academy.by / JC1-110-22 / Anastasia Melnikova / Done tasks 1-8

### PR DESCRIPTION
**Pull request by Stellmaria**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/test/java/com/rakovets/course/java/core/practice/concurrency/ProducerDemo.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.rakovets.course.java.core.practice.concurrency;
+
+import java.util.Queue;
+import java.util.concurrent.LinkedBlockingQueue;
+
+public class ProducerDemo {
+    public static void main(String[] args) throws InterruptedException {
+        Queue<Integer> queue = new LinkedBlockingQueue<>();
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/concurrency/ProducerDemo.java:8]: Dirty hack

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/Consumer.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                output.flush();
+            }
+        } catch (FileNotFoundException e) {
+            LOGGER.log(Level.SEVERE, e.getMessage());
+        } catch (IOException e) {
+            LOGGER.log(Level.WARNING, e.getMessage());
+        } catch (InterruptedException e) {
+            LOGGER.log(Level.INFO, e.getMessage());
+            Thread.currentThread().interrupt();
+        } catch (NullPointerException exception) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/Consumer.java:53]: Bad practice

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/Consumer.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    /**
+     * Consumer thread that takes numbers from the Queue and sleeps for a number of seconds equal to this number.
+     * In this case, the Consumer thread is sleeping in the order in which the integer values were entered by the user.
+     */
+    @Override
+    public void run() {
+        try (OutputStreamWriter output = new OutputStreamWriter(
+                new FileOutputStream(Path.of("resources", "text", "test.txt").toFile()))) {
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/Consumer.java:28]: Best practice: getting filename from CLI arguments

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/Producer.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public void run() {
+        while (true) {
+            String nextLine = SCANNER.nextLine();
+            try {
+                if ("-1".equals(nextLine)) {
+                    queue.add(Integer.valueOf(nextLine));
+                    break;
+                } else if (!(isNumeric(nextLine))) {
+                    throw new UserInputException("Incorrect input");
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/concurrency/Producer.java:35]: bad practice

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

File: src/test/java/com/rakovets/course/java/core/practice/concurrency/ProducerDemo.java (hunk 1 above)
Comment (Stellmaria) [reply src/test/java/com/rakovets/course/java/core/practice/concurrency/ProducerDemo.java:8]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/Consumer.java (hunk 2 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency/Consumer.java:53]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/Consumer.java (hunk 3 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency/Consumer.java:28]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/concurrency/Producer.java (hunk 4 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/concurrency/Producer.java:35]: fixed

**Pull request closed without merging**